dev/com.ibm.ws.jdbc/src/com/ibm/ws/jdbc/internal/JDBCDriverService.java

```diff
@@ -1037,6 +1078,189 @@ protected void setSharedLib(Library lib) {
             Tr.debug(this, tc, "setSharedLib", lib);
         sharedLib = lib;
     }
+    
+    @FFDCIgnore(Exception.class)
+    private void setupOracleLogging() {
+        final String method = "setupCustomOracleLogging";
+        
+        final String
+        ORACLELOG_ENABLE_TRACE = "oracle.jdbc.Trace",
+        ORACLELOG_FILE_SIZE_LIMIT = "oracleLogFileSizeLimit",
+        ORACLELOG_FILE_COUNT = "oracleLogFileCount",
+        ORACLELOG_FILENAME = "oracleLogFileName",
+        ORACLELOG_TRACELEVEL = "oracleLogTraceLevel",
+        ORACLELOG_FORMAT = "oracleLogFormat",
+        ORACLELOG_PACKAGENAME = "oracleLogPackageName";
+        
+        Logger parentLogger = Logger.getLogger(ORACLELOG_PARENT_PACKAGENAME);
+        
+        //If the 'oracle' logger isn't using a parent then it's possible the user configured their 
+        //application to send logs somewhere else.  Don't overwrite them.
+        if(!parentLogger.getUseParentHandlers()) {
+            return;
+        }
+        
+        // Fail fast: If some, but not all, required properties were provided tell user and return.
+        if( (System.getProperty(ORACLELOG_FILENAME) != null) == (System.getProperty(ORACLELOG_PACKAGENAME) != null) &&
```

Comment:
As currently written, this config -Doracle.jdbc.Trace=false -DoracleLogPackageName=oracle will cause the ORACLE_TRACE_ENABLE_INFO message to trigger even though trace is off, which it shouldn't.

Reply:
Correct, there should be an extra check here to only inform users if oracle trace was enabled, but not all properties for custom logging were set.